packages/ethers5/src/client.ts

```diff
@@ -169,7 +171,38 @@ export class Web3Modal extends Web3ModalScaffold {
           onUri(uri)
         })
 
-        await WalletConnectProvider.connect()
+        if (siweConfig?.options?.enabled) {
+          const { SIWEController } = await import('@web3modal/siwe')
+          const result = await WalletConnectProvider.authenticate({
+            nonce: await siweConfig.getNonce(),
+            methods: OPTIONAL_METHODS,
+            ...siweConfig.options.messageParams
+          })
+          // Auths is an array of signed CACAO objects https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-74.md
+          const signedCacao = result?.auths?.[0]
+          if (signedCacao) {
+            const { p, s } = signedCacao
+            const chainId = getDidChainId(p.iss)
+            const address = getDidAddress(p.iss)
+            if (address && chainId) {
+              SIWEController.setSession({
+                address,
+                chainId: parseInt(chainId, 10)
+              })
+            }
+            // Kicks off verifyMessage and populates external states
+            const message = WalletConnectProvider.signer.client.formatAuthMessage({
+              request: p,
+              iss: p.iss
+            })
+            SIWEController.verifyMessage({
+              message,
+              signature: s.s
+            })
+          }
```

Comment:
nit: Might be good to abstract this to a utils function like `verifyCacao` or similar